pkg/apis/provisioning/v1alpha1/provisioner.go

```diff
@@ -41,8 +61,24 @@ type ProvisionerSpec struct {
 	// launched by the Provisioner. If unspecified, supports all types. Cannot
 	// be specified if label "node.kubernetes.io/instance-type" is specified.
 	InstanceTypes []string `json:"instanceTypes,omitempty"`
```

Comment:
Do we need to guarantee this list is sorted? I'm just wondering whether we could wind up in a situation where somehow one pod has the list `{"m5.large", "c5.large"}` and wind up in a different group from a pod with the list `{"c5.large", "m5.large"}` (say)?

Reply:
I'm not sure I agree. These are at the provisioner level not the pod level.

Reply:
I think I see what you're saying. IIUC, this ordering will always be deterministic for any given pod.